Literal keys contents / values (#1036)
unicode-org/message-format-wg: spec/syntax.md

- _Duplicate Variant_:
  Each _variant_ MUST use a list of _keys_ that is unique from that
  of all other _variants_ in the _message_.

>   _Literal_ _keys_ are compared by their values, not their syntactical appearance.

Reviewer: ```suggestion
  _Literal_ _keys_ are compared by their _string values_, not their syntactical appearance.
```

The value of each _literal_ _key_ MUST be treated as if it were in
[Unicode Normalization Form C](https://unicode.org/reports/tr15/) ("NFC").

> Two _literal_ _keys_ are considered equal if their values are canonically equivalent strings,

Reviewer: ```suggestion
Two _literal_ _keys_ are considered equal if their _string values_ are canonically equivalent strings,
```

An **_<dfn>unquoted literal</dfn>_** is a _literal_ that does not require the `|`
quotes around it to be distinct from the rest of the _message_ syntax.

> An _unquoted literal_ MAY be used when the value of the _literal_

Reviewer: ```suggestion
An _unquoted literal_ MAY be used when the _string value_ of the _literal_
```

quoted-literal   = "|" *(quoted-char / escaped-char) "|"
unquoted-literal = 1*name-char
```
The **_<dfn>string value</dfn>_** of a _literal_ 
for _unquoted literals_ is the text content of that _literal_;
or for _quoted literals_, the text content of that _literal_ 
after removing the enclosing `|` characters

> then unescaping any escaped characters.

Reviewer: This feels a bit clumsy, but not objectionably so.

quoted-literal   = "|" *(quoted-char / escaped-char) "|"
unquoted-literal = 1*name-char
```
The **_<dfn>string value</dfn>_** of a _literal_ 
for _unquoted literals_ is the text content of that _literal_;
or for _quoted literals_, the text content of that _literal_ 
after removing the enclosing `|` characters

> and unescaping any escaped characters

Reviewer 2: ```suggestion
```

quoted-literal   = "|" *(quoted-char / escaped-char) "|"
unquoted-literal = 1*name-char
```

> The **_<dfn>value of a literal</dfn>_** is the text after removing the terminating | characters if the literal is quoted,

Reviewer: ```suggestion
The **_<dfn>string value</dfn>_** of a literal is the text after removing the terminating `|` characters if the literal is quoted,
```

Reviewer 2: Terminating doesn't seem right, since one of them is before the string starts. Literal is a term. Perhaps:

```suggestion
The **_<dfn>string value</dfn>_** of a _literal_ is the text content of that _literal_
not including any enclosing `|` characters of a _quoted literal_
and after unescaping any escaped characters.
```

Author: I like that phrasing better. However, my wording was a bit off; unquoted literals don't contain escaped characters. So probably clearer as the following:

The **_<dfn>string value</dfn>_** of a _literal_ is:
- for _unquoted literals_,  the text content of that _literal_
- for _quoted literals_,  text content of that _literal_, after removing the enclosing `|` characters and unescaping any escaped characters.

Reviewer: It's true that unquoted literals cannot contain escaped characters, but that just means that "any escaped characters" is always an empty set, as it'll also be in most cases for quoted literals as well.

I'm also not opposed to forking the definition as proposed.

Author: I think that formally we are safe because \ can't be in unquoted literals, but I like splitting it out.

Reviewer 2: ```suggestion
The **_<dfn>string value</dfn>_** of a _literal_ 
for _unquoted literals_ is the text content of that _literal_;
or for _quoted literals_, the text content of that _literal_ 
after removing the enclosing `|` characters
and unescaping any escaped characters
```
